### DISCUSSION TITLE
Implement the ability to use a dynamic 'incoming port' for rtorrent - to use with Gluetun

### DISCUSSION THREAD

**Discussion by notDavid** [crazy-max/docker-rtorrent-rutorrent#335] [Q&A]
### Description

[Gluetun](https://github.com/qdm12/gluetun) has [added the ability](https://github.com/qdm12/gluetun/issues/1488) to use Port Forwarding for VPN providers, documented [here](https://github.com/qdm12/gluetun-wiki/blob/29e3dee5024d82218e3f170b32f027102ff84aab/setup/advanced/vpn-port-forwarding.md)

Would be really cool if we could set `RT_INC_PORT` dynamically, by using this functionality.
The (random) opened port number by the VPN provider can be obtained via the Control Server. For example: `curl 192.168.1.5:8000/v1/openvpn/portforwarded`

**Comment by stickz** [comment]
It's required to restart `rTorrent` anyways when changing the port. Is it possible to restart the docker container with that environment variable? There should be a graceful exit, assuming your configuration works properly.

**Comment by notDavid** [comment]
Hi @stickz , thank you for your reply;

I just noticed [this post](https://github.com/qdm12/gluetun/issues/1488#issuecomment-1873417075), but i'm not sure how to implement it. Would this clarify your question / help in finding a solution?

> AFAIK the server unfortunately does not provide any way to get the port when using Wireguard, so you can use the temp file instead as a workaround.
> I currently have this set up with a scheduled cat command in my rtorrent.rc and it works fine :)
